src/test/java/com/github/firmwehr/gentle/testutil/EqualityChecker.java

```diff
@@ -0,0 +1,88 @@
+package com.github.firmwehr.gentle.testutil;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Method;
+import java.lang.reflect.RecordComponent;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+
+@SuppressWarnings("ClassCanBeRecord")
+public class EqualityChecker<T> implements Predicate<T> {
+	private static final MethodHandles.Lookup LOOKUP = MethodHandles.publicLookup();
+	private static final Map<Class<?>, List<MethodHandle>> ACCESSOR_CACHE = new HashMap<>();
+	private final T other;
+	private final Set<Class<?>> exceptTypes;
+	private final boolean deep;
+
+	public EqualityChecker(T other, Set<Class<?>> exceptTypes, boolean deep) {
+		this.other = other;
+		this.exceptTypes = exceptTypes;
+		this.deep = deep;
+	}
+
+	@Override
+	public boolean test(T t) {
+		return isEqual(t);
+	}
+
+	public boolean isEqual(T t) {
+		if (this.other == t) {
+			return true;
+		}
+		if (t == null) {
+			return false;
+		}
+		var handles = ACCESSOR_CACHE.computeIfAbsent(this.other.getClass(), EqualityChecker::createAccessors);
+		for (MethodHandle handle : handles) {
+			if (this.exceptTypes.contains(handle.type().returnType())) {
+				continue; // ignore this one
+			}
+			if (!isEqualAttribute(handle, t)) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	private boolean isEqualAttribute(MethodHandle handle, T t) {
+		Object otherAttribute;
+		Object rAttribute;
```

Comment:
The naming is a bit confusing. Why is this called `rAttribute` and the parameter `t`? Maybe consistently using `other` and `own` or sth like that would be nicer?